tauri-api/src/rpc.rs

```diff
@@ -22,9 +79,30 @@ use serde_json::Value as JsonValue;
 /// let cb = format_callback("callback-function-name", serde_json::to_value(&MyResponse {
 ///   value: "some value".to_string()
 /// }).expect("failed to serialize"));
-/// assert!(cb.contains(r#"window["callback-function-name"]({"value":"some value"})"#));
+/// assert!(cb.contains(r#"window["callback-function-name"](JSON.parse('{"value":"some value"}'))"#));
 /// ```
 pub fn format_callback<T: Into<JsonValue>, S: AsRef<str>>(function_name: S, arg: T) -> String {
+  let as_str = {
```

Comment:
Why have this? Since this just returns `as_str` at the end anyways, you could get rid of it and save a few lines.

Reply:
It implicitly drops `json_value` as it's not needed outside of that block.

Reply:
`json_value` actually stores the value of the JSON inside the enum, so if it's not dropped it could end up being a horrible waste of memory. Upon review, I am actually about to improve the memory usage a bit in the next commit.

Reply:
It would get dropped at the end of the function anyways though. Wouldn't that be fine?

Reply:
Nope, if you want to push a massive string, like a few GB, it'd end up duplicating in memory for no reason.

Demonstration that even on release mode, it doesn't aggressively drop stuff. I would imagine the compiler assumes the value is most likely not going to be big enough to aggressively drop. Uncomment the explicit drop to see the difference.

https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=b8b5df6933a792c03f40e221c29084ed

It'd be much better if you could explicitly consume the value into a string. Maybe I'll look into it a bit more.

Reply:
Yeah there doesn't seem to be a way to consume it into a string yet, unfortunately. This is the best we can do.

Reply:
Okay, sounds good then :+1:.